libs/island-ui/core/src/lib/DatePicker/DatePicker.tsx

```diff
@@ -140,50 +124,118 @@ export const DatePicker: React.FC<DatePickerProps> = ({
             setDatePickerState('closed')
             handleCloseCalendar && handleCloseCalendar(startDate)
           }}
-          onChange={(date: Date) => {
+          onChange={(date: any) => {
             setStartDate(date)
             handleChange && handleChange(date)
           }}
+          startDate={startDate}
+          required={required}
           customInput={
             <CustomInput
-              value={value}
-              onClick={onInputClick}
+              name={inputName}
+              label={label}
+              fixedFocusState={datePickerState === 'open'}
+              hasError={hasError}
+              errorMessage={errorMessage}
               placeholderText={placeholderText}
+              onInputClick={onInputClick}
             />
           }
-          renderCustomHeader={({ date, decreaseMonth, increaseMonth }) => {
-            const month = format(date, 'MMMM', { locale: getLocale(locale) })
-            const capitalizedMonth = `${month
-              .charAt(0)
-              .toUpperCase()}${month.slice(1)}`
+          renderCustomHeader={(props) => (
+            <CustomHeader locale={locale} {...props} />
+          )}
+        />
+      </div>
+    </div>
+  )
+}
+
+const CustomInput = ({
+  className,
+  placeholderText,
+  onInputClick,
+  fixedFocusState,
+  ...props
+}: InputProps & {
+  placeholderText?: string
+  onInputClick?: ReactDatePickerProps['onInputClick']
+}) => (
+  <Input
+    {...props}
+    icon="calendar"
+    iconType="outline"
+    fixedFocusState={fixedFocusState}
+    placeholder={placeholderText}
+  />
+)
+
+const getLocale = (locale: Locale) => {
+  return locale === 'is' ? is : locale === 'pl' ? pl : en
```

Comment:
maybe do something where it would be easier to add a new language? b/c you never know!

Reply:
I'll make this more reuseable